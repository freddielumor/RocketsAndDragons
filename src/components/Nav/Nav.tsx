import * as React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { appStyles } from '../../styles/appStyles'
import { media } from '../../styles/mediaQueries'

export const AppNav: React.FC = () => (
  <Nav>
    <NavLink to="/">Rockets</NavLink>
    <NavLink to="/dragons">Dragons</NavLink>
  </Nav>
)

/* -- Components -- */

// Nav
export const Nav = styled.nav`
  display: flex;
  justify-content: space-evenly;
  margin: auto;
  flex-direction: column;

  @media ${media.mobile} {
    flex-direction: row;
  }
`

// Nav Link
export const NavLink = styled(Link)`
  flex-direction: column;
  min-width: 6rem;
  margin-top: 0;
  padding: 1rem 1rem;
  background-color: ${appStyles.color.primary};
  color: ${appStyles.color.white};
  border-radius: 5px;

  &:not(:first-child) {
    margin-top: 1rem;
  }

  &:hover {
    background-color: ${appStyles.color.primaryHover};
  }

  @media ${media.mobile} {
    &:not(:first-child) {
      margin-top: 0;
    }
  }
`
