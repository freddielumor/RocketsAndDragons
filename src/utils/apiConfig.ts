// Congigure Endpoints For Data Fetching
const BASE_URL = 'https://api.spacexdata.com/v3'

export const ROCKETS_DATA_ENDPOINT = `${BASE_URL}/rockets`
export const DRAGONS_DATA_ENDPOINT = `${BASE_URL}/dragons`
