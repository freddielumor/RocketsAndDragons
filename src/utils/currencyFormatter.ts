// Format currency helper
export const formatCurrency = (value: number): string => {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
  }).format(value)
}
