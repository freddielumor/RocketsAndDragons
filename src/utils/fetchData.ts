import axios from 'axios'

// Data Fetching Utility
export default async function fetchData(url: string): Promise<any> {
  try {
    const response = await axios.get(url)
    return response
  } catch (error) {
    return error
  }
}
