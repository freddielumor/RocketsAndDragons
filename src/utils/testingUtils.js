import * as React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { createStore, applyMiddleware } from 'redux'
import { Provider as ReduxProvider } from 'react-redux'
import thunk from 'redux-thunk'
import { render } from '@testing-library/react'
import rootReducer from '../redux/reducers/index'
import '@testing-library/jest-dom'

// TESTING UTILITY

// This util alllows us to replicate the application state for testing purposes.
// We export a render function which contains the application state, to allow us to test our components in isolation their integration across the app.

// 1. Setup initial state and mock store
const initialState = {}
const mockStore = createStore(rootReducer, initialState, applyMiddleware(thunk))

// 2. Create a test provider containing the state & router
const testWrapper = ({ children }) => {
  return (
    <Router>
      <ReduxProvider store={mockStore}>{children}</ReduxProvider>
    </Router>
  )
}

// 3. Create the test render function which is used in our test files
const testRender = (ui) => render(ui, { wrapper: testWrapper })

// 4. re-export everything from Testing Library
export * from '@testing-library/react'

// 5. Override render method
export { testRender as render }
