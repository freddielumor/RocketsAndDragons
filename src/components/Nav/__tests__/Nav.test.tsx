import * as React from 'react'
import { render } from '../../../utils/testingUtils'
import 'jest-styled-components'

import { AppNav, Nav, NavLink } from '../Nav'

// Nav
it('renders Nav correctly', () => {
  const { container } = render(
    <AppNav>
      <Nav>
        <NavLink to="/test">Test</NavLink>
      </Nav>
    </AppNav>,
  )

  expect(container).toMatchSnapshot()
})
