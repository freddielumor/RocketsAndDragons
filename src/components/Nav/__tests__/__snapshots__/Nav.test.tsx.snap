// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Nav correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  margin: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 6rem;
  margin-top: 0;
  padding: 1rem 1rem;
  background-color: #f54e75;
  color: #ffffff;
  border-radius: 5px;
}

.c1:not(:first-child) {
  margin-top: 1rem;
}

.c1:hover {
  background-color: #b70a32;
}

@media (min-width:480px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:480px) {
  .c1:not(:first-child) {
    margin-top: 0;
  }
}

<div>
  <nav
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      Rockets
    </a>
    <a
      class="c1"
      href="/dragons"
    >
      Dragons
    </a>
  </nav>
</div>
`;
